tao_compiler/mlir/disc/transforms/disc_comp_intens_fusion_to_cuda_source.cc

```diff
@@ -0,0 +1,631 @@
+// Copyright 2022 The BladeDISC Authors. All rights reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "mlir-hlo/Dialect/lhlo/IR/lhlo_ops.h"
+#include "mlir-hlo/Dialect/mhlo/IR/hlo_ops.h"
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/Pass/Pass.h"
+#include "tensorflow/compiler/mlir/disc/IR/lhlo_disc_ops.h"
+#include "tensorflow/compiler/mlir/disc/disc_util.h"
+#include "tensorflow/compiler/mlir/disc/transforms/PassDetail.h"
+#include "tensorflow/compiler/mlir/disc/utils/source_emitter.h"
+
+namespace mlir {
+namespace disc_ral {
+namespace {
+
+constexpr const char* kSpecializedClassName = "SpecializedGemmFusion";
+constexpr const char* kSpecializedEpilogue = "SpecializedEpilogue";
+constexpr const char* kEpilogueIsHeavy = "EpilogueIsHeavy";
```

Comment:
nit: good to have those patterns with spacial prefix and suffix like `__EpilogueIsHeavy__` or `{EpilogueIsHeavy}` to distinguish them from normal keyword and variable names in source code.